### PR TITLE
Improve consistency with genetics stuff

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

Fixes a bunch of mistakes and inconsistencies with genetics stuff:
- ensuring all the mutations can actually use the chromosomes they're supposed to.
  - pretty much all abilities were set to use sync instead of energy.. even tho energy is what affects cooldowns???
  - in fact, some mutations had code to use some chromosomes... but didn't actually have the var set that'd let you apply said chromosome to it. (to be specific: cat claws and strengthened wings had this issue)
- added chromosome interactions to a few mutations that didn't use them before
  - shock touch, cryokinesis, chameleon: power
  - acid hands: power, sync
- upgrading a dna scanner will update the console's remaining cooldowns accordingly
- cat claws changes your attack sound to an appropiate slashing noise
- lots of code cleanup that isn't visible in-game

## Why It's Good For The Game

Consistency is good. Things working as intended is good.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://youtu.be/UFRSdASpg7k

</details>

## Changelog
:cl:
fix: All action mutations now properly take energy chromosomes (which reduce ability cooldowns) rather than synchronizer chromosomes.
fix: Cat Claws and Strengthened Wings now properly accept power chromosomes — they already had code that took the power chromosome into account, but couldn't actually have said chromosome applied to them.
add: Shock touch, cryokinesis, and chameleon can now accept power chromosomes to improve their respective powers.
add: Acid hands can take a power or sync chromosome, to increase the acid power or decrease the blood loss respectively.
soundadd: Cat Claws now changes your attack sound to slashing noises, similar to moths or lizards.
tweak: Upgrading a DNA scanner will accordingly update any cooldowns (joker and injector) for linked genetics consoles.
add: Inserting a genetics diskette into a genetics console will now actually mark any mutations on said disk as discovered. This likely won't change anything during actual rounds, but this will be useful during debugging or perhaps adminbus.
tweak: You won't waste shock touch charges accidentally clicking on the floor or a wall anymore.
code: Cleaned up the code of a lot of mutations quite a bit.
/:cl:

### CODE REVIEW

**Comment by DrDuckedGoose** [comment]
Like & Subscribe

**Comment by Absolucy** [comment]
@DrDuckedGoose ?

**Comment by Fronsis** [comment]
How are the values of acid/shock/cryo affected by the chromosomes? In terms of damage

I guess the chamaleon power up increases the speed of becoming invisible?

**Comment by Absolucy** [comment]
@Fronsis:
- **Acid Hands**: acid volume is multiplied by power, blood cost is multiplied by synchronizer
- **Shock Touch**: shock damage multiplied by power
- **Cryokinesis**: Temperature of beam multiplied by power
- **Chameleon**: alpha reduction per tick multiplied by power (so yes, become invisible faster)

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -117,6 +117,10 @@
 			: FALSE)\
 		: FALSE)
 #define HAS_TRAIT_NOT_FROM(target, trait, source) (target.status_traits ? (target.status_traits[trait] ? (length(target.status_traits[trait] - source) > 0) : FALSE) : FALSE)
+/// Adds a trait, with the source being the caller's type.
+#define ADD_TYPED_TRAIT(target, trait)	ADD_TRAIT(target, trait, "[type]")
```
Comment (PowerfulBacon) [inline code/__DEFINES/traits.dm]: Didn't another PR add this define? Same review applies from that to this in that variables are pulled from a non-explicit source

Comment (Absolucy) [reply code/__DEFINES/traits.dm]: Ah yeah, holopara PR did this, I forgot that this PR did too

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**